File: atcoder/abc126_c.cpp
#include<bits/stdc++.h>
using namespace std;
int main() {
    int n, k; 
    scanf("%d%d", &n, &k);
    double ans = 0.0;
    // cout << ans <<"\n";
    if (n >= k) {
        ans += 1.0*(n-k+1)/n;
    }
    for (int i = 1; i < min(n+1, k); ++i) {
        ans += 1.0/(1.0*n) * pow(0.5, (int) log2((k-1)/i) + 1);
    }
    printf("%.12f",ans);
}

File: training/OnlineJudge/atcoder/abc126_c.py
import math
n, k = map(int, input().split())
ans = 0
if (n >= k):
    ans += (n-k+1)/n
for i in range(1, min(n, k) + 1):
    ans += (1/n) * pow(0.5, int(math.log2((k-1)/i)))
print(ans)